service/controller/resource/masters/create.go

```diff
@@ -63,10 +72,5 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "no state change")
 	}
 
-	// We should wait and avoid further resources to reconciliate as long as the masters are not successfully deployed.
```

Comment:
Not sure I understand how this changes the semantics. It will continue to next resource (instance) if there hasn't been a state change?

Reply:
This is from my change. With this code the reconciliation didn't go further until master was upgraded. If there's a problem with VPN connection during that, it prevents master upgrade but operator cannot reconcile the broken connection. Therefore I changed to logic so that `instance` resource checks for `masters` resource's status condition and skips execution until it's `DeploymentCompleted`. For intended purpose it works the same but allows reconciliation of e.g. vpn resources (which don't change with flatcar but is crucial for whole operations).